xbmc/cores/VideoPlayer/DVDCodecs/Overlay/DVDOverlayCodecSSA.cpp

```diff
@@ -18,13 +18,17 @@
 
 #include <memory>
 
+#include "system.h"
 
-CDVDOverlayCodecSSA::CDVDOverlayCodecSSA() : CDVDOverlayCodec("SSA Subtitle Decoder")
+CDVDOverlayCodecSSA::CDVDOverlayCodecSSA()
+  : CDVDOverlayCodec("SSA Subtitle Decoder"), m_libass(std::make_shared<CDVDSubtitlesLibass>())
 {
   m_pOverlay = NULL;
-  m_libass   = NULL;
   m_order    = 0;
   m_output   = false;
+
+  if (!m_libass)
```

Comment:
Technically this isn't needed, but it's good defensive programming

Reply:
if the `make_shared` call fails what happens?

Reply:
it throws an out-of-memory exception. that's why we don't check for nullptr after `new`